### ISSUE TITLE
Transclusions: track selectors used in components so they can be checked against when the component is used

### ISSUE THREAD

**Issue by MichaelRFairhurst** [dart-lang/angular_analyzer_plugin#126]
Part of larger ticket #48

This can be done as its own pass over the html parse tree, since we're just looking for all `<ng-content>` tags with selectors.

Note that if the directive has an `<ng-content>` tag _without_ a specified selector, we should track that too. Then if we go to validate that all transcluded content is valid we can know in this case that we shouldn't bother checking.

Also note that this step could have a cyclic dependency...for example, a tree with two components.

```
@Component(
  selector: 'recursive-a',
  template: '<recursive-b *ngFor="item in tree" [item]="item"><b transcluded>...</b></recursive-b><ng-content selector="[transcluded]"></ng-content>',
  directives: [RecursiveB])
class RecursiveA {
  ...
}

@Component(
  selector: 'recursive-b',
  template: '<recursive-a [items]="item.subItems"><b transcluded>...</b></recursive-a><ng-content selector="[transcluded]></ng-content>",
  directives: [RecursiveA])
class RecunsiveB {
  ...
}
```

In this case, these may be in separate dart and/or html files. This could mean that if the transclusion selectors of directive A are part of the definition of directive A, and same with B, then importing A depends on B which depends on A.

Actually, not sure if this is supported regardless of transclusion...have to check. Though certainly adding the separate html files to the mix makes things more complicated.

This is P1 because it can allow us to track transclusions which are a big missing validation point, and can be a source of custom elements that otherwise aren't described, therefore causing false errors right now.

**Comment by zoechi** [comment]
There is also the `ngProjectAs` https://github.com/dart-lang/angular2/blob/61fcbf1b46286c7684d10699b7a92883c3c92d16/lib/src/compiler/template_parser.dart#L398
I'm not 100% sure it's supported in Dart. I saw it mentioned a few times in TS land but never tried it in Dart.

**Comment by MichaelRFairhurst** [comment]
Thanks Gunter! I should give ngProjectAs its own ticket, I think it has implications beyond here.

Also lowering the importance on this, transclusion support was bumped up in priority over a not fully thought out thing, I'm currently thinking.

**Comment by MichaelRFairhurst** [comment]
fixed in #230
